web/views.py

```
@@ -6955,3 +6973,448 @@ def delete_post(request, post_id):
+
+    context = {
+        "membership": membership,
+        "invoices": invoices,
+        "events": events,
+    }
+
+    return render(request, "membership/manage.html", context)
+
+
+@login_required
+def membership_cancel(request) -> HttpResponse:
```

Ruff (0.8.2)

Missing type annotation for function argument `request`

(ANN001)